EIPS/eip-6672.md

```diff
@@ -0,0 +1,247 @@
+
+### Key Choices for Redemption Flag
+
+The combination of `_operator`, `_tokenId`, and `_redemptionId` is REQUIRED to be used as the key in the redemption flag key-value pairs. The reasoning is as follows:
+
+- Operator wallet address, i.e. `_operator`
+
+    It's possible that there are more than one party who would like to use the same NFT for redemption. For example, MisterPunks NFTs are eligible to be redemeed for both Event-X and Event-Y tickets, and each event's ticket redemption is handled by a different operator.
+
+- Token identifier, i.e. `_tokenId`
+
+    Each NFT holder will have different redemption records created by the same operator. Therefore, it's important to use token identifier as one of the keys.
+
+- Redemption identifier, i.e. `_redemptionId`
+
+    Using `_redemptionId` as one of the keys enables NFT holders to redeem the same NFT to the same operator in multiple campaigns. For example, Operator-X has 2 campaigns, i.e. campaign A and campaign B, and both campaigns allow for MisterPunks NFTs to be redemeed for physical action figures. Holder of MisterPunk #7 is eligible for redemption in both campaigns and each redemption is recorded with the same `_operator` and `_tokenId`, but with different `_redemptionId`.
+
+### Metadata Choices (metadata extension)
+
+To provide additional information about a redemption, `redemptions` key-value pairs is added to the "[ERC-721](./eip-721.md) Metadata Extension". The key format for the redemptions key-value pairs MUST be standardized as `operator-tokenId-redemptionId`, where `operator` is the operator wallet address, `tokenId` is  the identifier of the token that has been redeemed, and `redemptionId` is the redemption identifier. The same combination of keys is used for the redemption flag, and the same rationale applies.
+
+The value of the key `operator-tokenId-redemptionId` is an object that contains the `status` and `description` of the redemption.
+
+- Redemption status, i.e. `status`
+
+    The redemption status can have a more granular level, rather than just being a flag with a `true` or `false` value. For instance, in cases of physical goods redemption, we may require the redemption status to be either `redeemed`, `paid`, or `shipping`. It is RECOMMENDED to use a string enum that is comprehensible by both the operator and the marketplace or any other parties that want to exhibit the status of the redemption.
+
+- Description of the redemption, i.e. `description`
+
+    The `description` can be used to provide more details about the redemption, such as information about the concert ticket, a detailed description of the action figures, and more.
+
+## Backwards Compatibility
+
+This standard is compatible with [ERC-721](./eip-721.md).
+
+## Reference Implementation
+
+Here's an example of an [ERC-721](./eip-721.md) that includes the Multi-Redeemable ([IERC-6672](./eip-6672.md)) extension:
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
```

Comment:
If the code is inline in the proposal itself, it must be licensed under `CC0-1.0`. We're a bit more permissive for files under `assets/`, but it still needs to be an open source license (like apache or mit.) `UNLICENSED` literally means no license, so we can't distribute it.

Reply:
set the license to `MIT` and put the contract under `assets/eip-6672/contracts/`